kernel/src/simulationTools/Simulation.cpp

```diff
@@ -544,16 +544,37 @@ void Simulation::prepareIntegratorForDS(SP::OneStepIntegrator osi,
                                         SP::DynamicalSystem ds,
                                         SP::Model m, double time)
 {
+  assert(m && m->nonSmoothDynamicalSystem() && "Simulation::prepareIntegratorForDS requires a Model with an NSDS.");
+
```

Comment:
The check m->nonSmoothDynamicalSystem() should result in a exception, since m->nonSmoothDynamicalSystem()->topology()->setOSI(ds, osi); is going to fail otw. I'd do
if (!m->nonSmoothDynamicalSystem()) { RAISE EXCEPTION }